contributors/design-proposals/network/pod-resolv-conf.md

```diff
@@ -0,0 +1,224 @@
+search foo.com bar.com
+options ndots:1
+```
+
+The pod will get the following `/etc/resolv.conf`:
+
+```bash
+nameserver 10.240.0.10
+# Comments only for explication purposes.
+#
+#      $(NAMESPACE).svc.$CLUSTER
+#      |                     my.dns.search.suffix
+#      |                     |                    $(HOST)
+#      |                     |                    |
+#      V                     V                    V
+search ns1.svc.cluster.local my.dns.search.suffix foo.com bar.com
+# Populated from custom.options
+options ndots:2
+```
+
+## More examples
+
+The following is a Pod dnsParams that only contains the host search paths:
+
+```yaml
+dnsParams:
+  custom:
+    searchPaths:
+    - $HOST
+```
+
+Override `ndots` and add custom search path. Note that overriding the ndot may
+break the functionality of some of the search paths the
+
+```yaml
+dnsParams:
+  custom:
+    searchPaths:
+    - my.custom.suffix
+    - $HOST
+    options:
+    - "ndots:3"
+```
+
+# API changes
+
+```go
+type PodSpec struct {
+	...
+	DNSPolicy string
+	DNSParams *PodDNSParams
+    ...
+}
+
+type PodDNSParams struct {
+ 	Custom PodDNSParamsCustom
+}
+
+type PodDNSParamsCustom struct {
+	Search []string
+ 	Options []string
+}
+
+// This will not appear in types.go but is here for explication purposes.
+type DNSParamsSubstitution string
+const (
+	DNSParamsSearchPathNamespace = "$(NAMESPACE)"
+	DNSParamsSearchPathClusterDomain = "$(CLUSTER)"
+	DNSParamsSearchPathHostPaths = "$(HOST)"
+)
+```
+
+## Semantics
+### searchPath
+
+If `dnsPolicy: Custom` is used, then the `search` line will be constructed from
+the entries listed in `dnsParams.custom.search`:
+
+```go
+// Note: pseudocode does not include input validation.
+func SearchPath(params *PodDNSParams) []string {
+	var searchPaths []string
+	for _, entry := range params.SearchPaths {
+		if entry == "$HOST" {
+			searchPaths = append(searchPaths, HostSearchPaths...)
+			break
+		}
+		searchPaths = append(searchPaths, expand(entry))
+	}
+	return searchPaths
+}
+
+func expand(s string) string {
+	labels := strings.Split(s, ".")
+	for i := range labels {
+		if labels[0] == "$" {
+			labels[i] = Substitute(labels[i])
+		}
+	}
+	return strings.Join(labels, ".")
+}
+
+var LabelSubstitutions map[string]string
+
+func Substitute(l label) string {
+  if s, ok := LabelSubstitutions[l]; ok {
+    return s
+  }
+  return l
+}
+```
+
+#### Substitutions
+
+`Substitute` will replace DNS labels that begin with `$` with values for the
+given Pod. Note: the DNS label MUST be an EXACT string match. E.g. if the pod
+namespace is `my-ns`, `abc$(NAMESPACE)1234` will NOT be expanded to
+`abcmy-ns1234`, but will be an invalid configuration.
+
+| Substitution | Description |
+| ----   | ---- |
+| `$(NAMESPACE)` | Namespace of the Pod |
+| `$(CLUSTER)` | Kubernetes cluster domain (e.g. `cluster.local`) |
+| `$(HOST)` | Host search paths. This is a special substitution that MUST ONLY appear at the end of the `searchPaths` list |
```

Comment:
If it can only be at the end why not just have a boolean `includeHostSearchPaths`?

Reply:
Making this default to adding the host search paths to make it easier for users:
`excludeHostSearchPaths`
Thoughts?